src/dataplane-protocol/src/record.rs

```diff
@@ -453,9 +431,28 @@ mod test {
         Ok(())
     }
 
-    /// test decoding of records when one of the batch was truncated
     #[test]
-    fn test_decode_batch_truncation() {
```

Comment:
I deleted the truncation stuff because it seems that it's a bad use case to take the bytes and modify them directy. Thoughts @sehz?

Reply:
LOL.  That is to  simulate where SPU truncates output bytes in order to comply with max bytes per response.  This is critical test to demonstrate receivers can recover much of batches even thought some of the batches are incomplete.

Reply:
Part of the reason I removed it was because I couldn't figure out how to change the truncation bytes. Given that the SPU truncates it, that'll also have to change (probably why the build is currently failing) so should we keep the encoder changes to `RecordSet` or not?

Reply:
We should make it changes only to `DefaultAsyncBuffer`